src/components/sde/tests/Makefile

```diff
@@ -18,26 +18,31 @@ FFLAGS +=-g
 CFLAGS +=-g
 CXXFLAGS +=-g -std=c++11
 
-ifeq ($(BUILD_LIBSDE_SHARED),yes)
-	LDFLAGS += -Llib -L$(datadir) -L$(datadir)/libpfm4/lib -lpapi -lpfm -lsde
+ifeq ($(BUILD_LIBSDE_STATIC),yes)
+	LD_STATIC_FLAGS = $(LDFLAGS) $(datadir)/libpapi.a $(datadir)/libsde.a
 	LIBSDE=yes
-else ifeq ($(BUILD_LIBSDE_STATIC),yes)
-	LDFLAGS += -Llib $(datadir)/libsde.a -lpapi
+endif
+
+ifeq ($(BUILD_LIBSDE_SHARED),yes)
+	sdeLDFLAGS = $(LDFLAGS) -Llib -L$(datadir) -L$(datadir)/libpfm4/lib -lpapi -lpfm -lsde
 	LIBSDE=yes
+else
+	sdeLDFLAGS = $(LD_STATIC_FLAGS)
 endif
 
 SDE_F08_API=../sde_F.F90
 
 ifeq ($(LIBSDE),yes)
 	TESTS = Minimal_Test Minimal_Test++ Simple_Test Simple2_Test Simple2_NoPAPI_Test Simple2_Test++ Recorder_Test Recorder_Test++ Created_Counter_Test Created_Counter_Test++ Overflow_Test Counting_Set_Simple_Test Counting_Set_MemLeak_Test Counting_Set_Simple_Test++ Counting_Set_MemLeak_Test++
-endif
+
 ifeq ($(BUILD_LIBSDE_STATIC),yes)
 	TESTS += Overflow_Static_Test
 endif
 
 ifeq ($(ENABLE_FORTRAN_TESTS),yes)
```

Comment:
As a small suggestion, but completely up to you if you want to apply the change is to indent this block of code for readability.